### ISSUE TITLE
Doesn't work with i18n 1.3.0

### ISSUE THREAD

**Issue by ghiculescu** [iGEL/it#26]
**Note: there's a fix for this, see comments below. Basically you need `i18n` version <= 1.2, or >= 1.4.**

-----------------------

This crashes:

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'rails', '5.1.6.1', require: false
  gem 'it', '1.0.0', require: false
  gem 'i18n', '1.3.0', require: false
end

require 'action_view' # `it` depends on this, need to require after gemfile?
require 'active_support/all' # `it` depends on this, need to require after gemfile?
require 'it'
require 'i18n'

puts "using I18n #{I18n::VERSION}"

I18n::Backend::Simple.include(I18n::Backend::Fallbacks)

I18n.backend.store_translations(:en, {test1: "test1 %{interp}"})
I18n.backend.store_translations(:"en-AU", {test1: "test1 %{interp}, mate"})
I18n.backend.store_translations(:en, {test2: "test2 %{b: %{interp}}"})
I18n.backend.store_translations(:"en-AU", {test2: "test2 %{b:%{interp}}, mate"})

puts I18n.t("test1", interp: "using I18n.t")
puts I18n.t("test1", interp: "using I18n.t", locale: "en-AU")
puts I18n.t("test2", interp: "using I18n.t")
puts I18n.t("test2", interp: "using I18n.t", locale: "en-AU")

puts It.it("test1", interp: "using It.it") # this crashes on i18n v 1.3.0
puts It.it("test2", interp: "using It.it", b: It.tag(:b))

=begin
using I18n 1.3.0
test1 using I18n.t
test1 using I18n.t, mate
test2 %{b: using I18n.t}
test2 %{b:using I18n.t}, mate
/Users/alex/.rvm/gems/ruby-2.3.8/gems/i18n-1.3.0/lib/i18n/config.rb:99:in `block in missing_interpolation_argument_handler': missing interpolation argument :interp in "test1 %{interp}" ({"fallback_in_progress"=>true} given) (I18n::MissingInterpolationArgument)
  from /Users/alex/.rvm/gems/ruby-2.3.8/gems/i18n-1.3.0/lib/i18n/interpolate/ruby.rb:31:in `block in interpolate_hash'
  from /Users/alex/.rvm/gems/ruby-2.3.8/gems/i18n-1.3.0/lib/i18n/interpolate/ruby.rb:23:in `gsub'
  from /Users/alex/.rvm/gems/ruby-2.3.8/gems/i18n-1.3.0/lib/i18n/interpolate/ruby.rb:23:in `interpolate_hash'
  from /Users/alex/.rvm/gems/ruby-2.3.8/gems/i18n-1.3.0/lib/i18n/interpolate/ruby.rb:19:in `interpolate'
=end

```

This works fine:

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'rails', '5.1.6.1', require: false
  gem 'it', '1.0.0', require: false
  gem 'i18n', '1.2.0', require: false
end

require 'action_view' # `it` depends on this, need to require after gemfile?
require 'active_support/all' # `it` depends on this, need to require after gemfile?
require 'it'
require 'i18n'

puts "using I18n #{I18n::VERSION}"

I18n::Backend::Simple.include(I18n::Backend::Fallbacks)

I18n.backend.store_translations(:en, {test1: "test1 %{interp}"})
I18n.backend.store_translations(:"en-AU", {test1: "test1 %{interp}, mate"})
I18n.backend.store_translations(:en, {test2: "test2 %{b: %{interp}}"})
I18n.backend.store_translations(:"en-AU", {test2: "test2 %{b:%{interp}}, mate"})

puts I18n.t("test1", interp: "using I18n.t")
puts I18n.t("test1", interp: "using I18n.t", locale: "en-AU")
puts I18n.t("test2", interp: "using I18n.t")
puts I18n.t("test2", interp: "using I18n.t", locale: "en-AU")

puts It.it("test1", interp: "using It.it")
puts It.it("test2", interp: "using It.it", b: It.tag(:b))

=begin
using I18n 1.2.0
test1 using I18n.t
test1 using I18n.t, mate
test2 %{b: using I18n.t}
test2 %{b:using I18n.t}, mate
test1 using It.it
test2 <b>using It.it</b>
=end

```

This also works fine:

```ruby
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'rails', '5.1.6.1', require: false
  gem 'it', '1.0.0', require: false
  gem 'i18n', '1.3.0', require: false
end

require 'action_view' # `it` depends on this, need to require after gemfile?
require 'active_support/all' # `it` depends on this, need to require after gemfile?
require 'it'
require 'i18n'

puts "using I18n #{I18n::VERSION}"

# note that we are not using fallbacks this time

I18n.backend.store_translations(:en, {test1: "test1 %{interp}"})
I18n.backend.store_translations(:"en-AU", {test1: "test1 %{interp}, mate"})
I18n.backend.store_translations(:en, {test2: "test2 %{b: %{interp}}"})
I18n.backend.store_translations(:"en-AU", {test2: "test2 %{b:%{interp}}, mate"})

puts I18n.t("test1", interp: "using I18n.t")
puts I18n.t("test1", interp: "using I18n.t", locale: "en-AU")
puts I18n.t("test2", interp: "using I18n.t")
puts I18n.t("test2", interp: "using I18n.t", locale: "en-AU")

puts It.it("test1", interp: "using It.it")
puts It.it("test2", interp: "using It.it", b: It.tag(:b))

=begin
using I18n 1.3.0
test1 using I18n.t
test1 using I18n.t, mate
test2 %{b: using I18n.t}
test2 %{b:using I18n.t}, mate
test1 using It.it
test2 <b>using It.it</b>
=end

```

So in summary, since upgrading to `i18n` 1.3, if **fallbacks** are enabled, `It.it` is crashing when attempting to do any interpolation.

**Comment by ghiculescu** [comment]
https://github.com/svenfuchs/i18n/pull/454 fixes this, at least for my use case.

**Comment by ghiculescu** [comment]
This is fixed in `i18n` 1.4 - https://github.com/svenfuchs/i18n/issues/452#issuecomment-450761851

So basically `it` works fine with `i18n` <= 1.2, or >= 1.4.

**Comment by ghiculescu** [comment]
I don't have any better ideas. Hopefully anyone wondering why the current version is broken will see this.
